src/foam/nanos/ruler/RuleEngine.java

```diff
@@ -29,7 +29,7 @@
   private X userX_;
 
   public RuleEngine(X x, X systemX, DAO delegate) {
-    setX(systemX);
+    setX(systemX.put("theme", x.get("theme")));
```

Comment:
Carry the current theme to system context so that we don't need to do this on every rules that need the system context and will also trigger notification.

Reply:
If the RuleEngine is first accessed from something like a CRON Job, then 'x' will be the systemX and it won't contain a theme. If it's accessed from the web, then it will just be stuck with the theme of the first user to access this service, which will be difficult to debug in the future.

Reply:
yes, if manually run a cron/script, then you'll get that users (admin) theme, which is not what we want for the rule run.

Reply:
An approach discussed earlier:  add spid to Theme and only support one Theme per spid.  The ThemeDomains can continue to handle the many domains per whilelabel - all pointing to the same Theme - mind you, no way to enforce this.

Reply:
No, RuleEngine is a new instance per execution, the subsequent won't be affected by the former.

Reply:
SPID's are meant to be hierarchical, so a company could have as many as they need if they want multiple themes.

Reply:
@kgrgreer I'm good with this approach.

Reply:
SPID is added to theme domain, this approach expects the users of a specific SPID to use/come from the same domain.

Reply:
Our current ServiceProvider implementation does not support hierarchy.

Reply:
If we just support one theme per spid then spid goes on Theme model.  We can still support Theme per group and this is independent of spid supporting hierarchy. 

web -> ThemeDomain -> theme
cron -> user (spid) [-> group] -> theme